plugins/inputs/mongodb/mongodb.go

```diff
@@ -164,6 +168,24 @@ func (m *MongoDB) Gather(acc telegraf.Accumulator) error {
 		wg.Add(1)
 		go func(srv *Server) {
 			defer wg.Done()
+			if !srv.reachable {
+				// is not reachable try a reconnect
+				disconnectCtx, disconnectCancel := context.WithTimeout(context.Background(), 1*time.Second)
+				defer disconnectCancel()
+				err := srv.client.Disconnect(disconnectCtx)
```

Comment:
`client` needs to be checked if it isn't nil before calling `Disconnect`, because `src.reachable` could be set to false when  `mongo.Connect` fails resulting in a nil `client`.